internal/resources/aws/rds_cluster_instance.go

```diff
@@ -44,25 +44,14 @@ func (r *RDSClusterInstance) BuildResource() *schema.Resource {
 		PurchaseOption: strPtr("on_demand"),
 	}
 
+	var err error
 	if r.ReservedInstanceTerm != nil {
-		resolver := reservedInstanceResolver{
-			term:          strVal(r.ReservedInstanceTerm),
-			paymentOption: strVal(r.ReservedInstancePaymentOption),
-			dbInstance:    true,
-		}
-		valid, err := resolver.Validate()
-		if err != "" {
-			log.Warnf(err)
-		}
-		if valid {
-			purchaseOptionLabel = "reserved"
-			priceFilter = &schema.PriceFilter{
-				PurchaseOption:     strPtr("reserved"),
-				StartUsageAmount:   strPtr("0"),
-				TermLength:         strPtr(resolver.Term()),
-				TermPurchaseOption: strPtr(resolver.PaymentOption()),
-			}
+		resolver := newRdsReservationResolver(strVal(r.ReservedInstanceTerm), strVal(r.ReservedInstancePaymentOption))
+		priceFilter, err = resolver.PriceFilter()
```

Comment:
we actually only want to explicitly set the priceFilter if err is not nil. Because if it isn't then priceFilter will be set to nil.

Reply:
Sorry copy and paste refuse! Thanks for the review!